proto/src/test/java/com/google/gson/protobuf/functional/ProtosWithMapTypesTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2010 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gson.protobuf.functional;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.protobuf.ProtoTypeAdapter;
+import com.google.gson.protobuf.ProtoTypeAdapter.EnumSerialization;
+import com.google.gson.protobuf.generated.Bag;
+import com.google.gson.protobuf.generated.Bag.SimpleProto;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.GeneratedMessage;
+import junit.framework.TestCase;
+
+public class ProtosWithMapTypesTest extends TestCase {
+  private Gson gson;
+
+  @Override
+  protected void setUp() throws Exception {
+    super.setUp();
+    gson = new GsonBuilder().registerTypeHierarchyAdapter(
+      GeneratedMessage.class, ProtoTypeAdapter.newBuilder()
+          .setEnumSerialization(EnumSerialization.NUMBER)
+          .build())
+      .create();
+  }
+
+  //TODO
```

Comment:
Do you have a test which shows that this works as expected?

Note that I am not a Maintainer, and it could happen that even if you change this pull request it might not be merged (in the near future).